lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -91,29 +91,75 @@ end)
 /******************************************************************************/
 -- Name functions
 
--- Set the name of the E2 itself
-e2function void setName( string name )
-	local e = self.entity
-	if (e.name == name) then return end
-	if (name == "generic" or name == "") then
+local CurTime = CurTime
+
+local antiSpamLookup = {}
+local function NameSpamCheck(ent) -- Returns true if it is being spammed; otherwise false
+	if antiSpamLookup[ent] and antiSpamLookup[ent] > CurTime() then
+		return true
+	end
+	antiSpamLookup[ent] = CurTime() + 0.1 -- Adjust the delay here, it is applied to setter (setName/setComponentName) functions
```

Comment:
This implementation (unlike the limits for holo, find and print) is just based on 1 name change every <timestep>. A bandwidth-based implementation would be better suited so people can still update the name frequently if it is short.